File: app/src/main/java/com/unitedbustech/eld/view/ClickProxy.java
package com.unitedbustech.eld.view;

import android.view.View;

/**
 * @author yufei0213
 * @date 2018/2/26
 * @description 点击事件代理方法，防止过快点击
 */
public class ClickProxy implements View.OnClickListener {

    private static final long CLICK_DURATION_LIMIT = 1000;

    private View.OnClickListener origin;

    private long lastClickTime = 0;

    public ClickProxy(View.OnClickListener origin) {

        this.origin = origin;
    }

    @Override
    public void onClick(View v) {

        if (System.currentTimeMillis() - lastClickTime >= CLICK_DURATION_LIMIT) {

            origin.onClick(v);
            lastClickTime = System.currentTimeMillis();
        }
    }
}

File: app/src/main/assets/js/utils/click-proxy.js
/**
 * @author yufei0213
 * @date 2018/2/27
 * @description 点击事件的代理方法
 */
var ClickProxy = function (func, funcScope) {

    this.CLICK_DURATION = 1000;

    this.lastClickTime = 0;

    this.func = func;
    this.funcScope = funcScope;
};

/**
 * 点击事件代理方法
 * @param func 方法
 * @param funcScope 方法作用域
 * @return false 禁用事件冒泡
 */
ClickProxy.prototype.click = function (event) {

    if ((new Date().getTime() - this.lastClickTime) > this.CLICK_DURATION) {

        this.func.call(this.funcScope, event);
        this.lastClickTime = new Date().getTime();
    }
};